### PR TITLE
build(dds): Fix type generation for ESM and add exports field

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR updates the majority of packages in /dds (merge-tree and sequence excluded) to generate proper types in ESM builds and adds an `exports` field. The packages were already building using moduleResolution: node16 by virtue of inheriting from the new shared configs, so no changes were needed there. They also have a test using arethetypeswrong that runs in CI.

### CODE REVIEW

File: fluidBuild.config.cjs [hunk 1]
```diff
@@ -23,7 +23,14 @@ module.exports = {
 			script: false,
 		},
 		"compile": {
-			dependsOn: ["commonjs", "build:esnext", "build:test", "build:copy"],
+			dependsOn: [
+				"commonjs",
+				"build:esnext",
+				"build:test",
+				"build:copy",
+				// "^build:rename-types",
```
Comment (Josmithr) [inline fluidBuild.config.cjs:31]: Leaving the same comment here for completion: let's make sure this gets cleaned up 😋

File: fluidBuild.config.cjs [hunk 2]
```diff
@@ -159,9 +167,17 @@ module.exports = {
 				// Can be removed once the policy handler is updated to support tsc-multi as equivalent to tsc.
```
Comment (Josmithr) [inline fluidBuild.config.cjs:167]: Nit: might be worth noting the work item number tracking this cleanup for reference.

File: packages/dds/test-dds-utils/package.json [hunk 3]
```diff
@@ -17,10 +17,10 @@
 	"scripts": {
 		"api": "fluid-build . --task api",
 		"api-extractor:commonjs": "api-extractor run --local",
-		"api-extractor:esnext": "copyfiles -u 1 \"dist/**/*-@(alpha|beta|public|untrimmed).d.ts\" lib",
```
Comment (Josmithr) [inline packages/dds/test-dds-utils/package.json:20]: We don't need the esnext build here?

**Review by Josmithr — APPROVED** [review]

File: packages/dds/test-dds-utils/package.json (hunk 3 above)
Comment (tylerbutler) [reply packages/dds/test-dds-utils/package.json:20]: This package is not building ESM. There's no build:esnext script.

Comment (Josmithr) [reply packages/dds/test-dds-utils/package.json:20]: No, I know. I guess I'm wondering why this package only wants a commonjs build. Probably not a concern for this PR, just wondering if this is something we need to track for later?

**Pull request merged**